### PR TITLE
fix: resolve CodeQL findings before develop→main promotion

### PR DESCRIPTION
**Pull request by ZappoMan**
## Summary

Addresses [CodeQL](https://github.com/Artificer-Innovations/BeakerStack/pull/395) and [code quality](https://github.com/Artificer-Innovations/BeakerStack/pull/395) review comments blocking the develop→main promotion in #395.

- **HTML sanitization** — Replace regex-based script/event-handler stripping in `@beakerstack/articles` and `@beakerstack/help` with DOMPurify + JSDOM at build time (same library already used at render time).
- **ReDoS** — Replace polynomial markdown link regex in `articleUtils` / `build-help` with linear `stripMarkdownLinks` parsing.
- **Lint** — Remove unused `pkgRoot` and path imports from `packages/connections/vitest.config.ts`.

## Test plan

- [x] `packages/articles` — `npm test` (71 tests)
- [x] `packages/help` — `npm test` (41 tests)
- [x] `packages/connections` — `npm test` (56 tests)
- [ ] CI CodeQL check passes on this PR
- [ ] Merge to `develop`, then re-run #395 promotion

### CODE REVIEW

File: packages/articles/src/sanitizeArticleHtml.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
+import DOMPurify from 'dompurify';
+
 /** Sanitize article HTML generated from trusted repo markdown at build time. */
 export function sanitizeArticleHtml(html: string): string {
-  return html
-    .replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '')
-    .replace(/\s(on\w+)=("[^"]*"|'[^']*'|[^\s>]+)/gi, '')
-    .replace(/href\s*=\s*["']javascript:[^"']*["']/gi, 'href="#"');
+  const window = new JSDOM('').window;
+  return DOMPurify(window).sanitize(html);
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/articles/src/sanitizeArticleHtml.ts]: Fixed in c2e635d — JSDOM and DOMPurify are now created once at module scope:

```ts
const purify = DOMPurify(new JSDOM('').window);

export function sanitizeArticleHtml(html: string): string {
  return purify.sanitize(html);
}
```

Build-time article sanitization reuses the same instance for every call.

File: packages/help/src/sanitizeHelpHtml.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { JSDOM } from 'jsdom';
+import DOMPurify from 'dompurify';
+
+/** Sanitize help HTML generated from trusted repo markdown at build time. */
+export function sanitizeHelpHtml(html: string): string {
+  const window = new JSDOM('').window;
+  return DOMPurify(window).sanitize(html);
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/help/src/sanitizeHelpHtml.ts]: Fixed in c2e635d — same pattern as articles: module-scoped `purify` instance so each help section during `build-help` does not allocate a new JSDOM window.

```ts
const purify = DOMPurify(new JSDOM('').window);

export function sanitizeHelpHtml(html: string): string {
  return purify.sanitize(html);
}
```

**Review by rahul-artificerinnovations — APPROVED** [review]
## Review — PR #396 (@rahul)

All three CodeQL/quality findings are correctly addressed. No blockers, no minors. **Approved.**

---

### ✅ HTML sanitization

**`sanitizeArticleHtml.ts` / `sanitizeHelpHtml.ts`** — Both now use `DOMPurify(new JSDOM('').window)` with a module-level instance (one instance per module, not per call — correct). DOMPurify is the right tool here; regex stripping was brittle and incomplete. ✅

**`jsdom` moved to `dependencies`** — Correct. The sanitizers are now called from build-time scripts (not test-only code), so jsdom must be a runtime dep, not a devDependency. ✅

**Sanitization tests** — Both packages cover script tags (including spaced `<script >`), event-handler attributes (`onerror`), and `javascript:` hrefs. The `sanitizeArticleHtml.test.ts` removal of `expect(output).toContain('href="#"')` is correct — DOMPurify 3.x strips the attribute entirely rather than replacing with `#`, which is strictly more secure. ✅

---

### ✅ ReDoS fix

**`stripMarkdownLinks`** — Linear O(n) single-pass parser replaces the polynomial link regex. The implementation is correct:
- Scans for `[label](url)` pattern with `indexOf`; each character visited at most once per scan pass
- Falls through without consuming on any partial match (`[` with no `]`, `]` with no `(`, `(` with no `)`)
- Malformed bracket edge cases tested in both `articleUtils.test.ts` and `build-help.test.ts`: unclosed `[`, `[not a link]` (no paren), `[open](no-close` (no closing paren)

The function is duplicated between `articleUtils.ts` and `build-help.ts` — acceptable given they're in separate packages with different build contexts. ✅

---

### ✅ Lint

**`packages/connections/vitest.config.ts`** — Unused `path`, `fileURLToPath`, and `pkgRoot` removed. ✅
